webapp/app/js/compare.js

```diff
@@ -0,0 +1,403 @@
+                rawResultsMap[data.series_signature].name = series.name;
+                rawResultsMap[data.series_signature].platform = series.platform;
+              });
+            });
+          })).then(function () {
+
+            // find summary results for all tests/platforms for the original rev
+            var signatureURL = thServiceDomain + '/api/project/' +
+                           $scope.newProject + '/performance-data/0/' +
+                           'get_performance_data/?interval_seconds=' + $scope.timeRange;
+
+            //ok, now get the new revision
+            var signatureListURL = thServiceDomain + '/api/project/' + $scope.newProject + 
+              '/performance-data/0/get_performance_series_summary/?interval=' +
+              $scope.timeRange;
+
+            var newRawResultsMap = {};
+            var newSeriesList = [];
+
+
+            $http.get(signatureListURL).then(function(response) {
+              Object.keys(response.data).forEach(function(signature) {
+                var seriesSummary = getSeriesSummary(signature,
+                                                     response.data[signature],
+                                                     optionCollectionMap,
+                                                     $stateParams.pgo,
+                                                     $stateParams.e10s);
+
+                if (seriesSummary != null && seriesSummary.signature !== undefined) {
+                  newSeriesList.push(seriesSummary);
+
+                  if ($scope.platformList.indexOf(seriesSummary.platform) === -1) {
+                    $scope.platformList.push(seriesSummary.platform);
+                  }
+
+                  if ($scope.testList.indexOf(seriesSummary.name) === -1) {
+                    $scope.testList.push(seriesSummary.name);
+                  }
+                }
+              });
+              $scope.testList.sort();
+              $scope.platformList.sort();
+
+
+              $q.all(newSeriesList.map(function(series) {
+                return $http.get(signatureURL + "&signatures=" + series.signature).then(function(response) {
+                  response.data.forEach(function(data) {
+                    newRawResultsMap[data.series_signature] = calculateStats(data.blob, $scope.newResultSetID);
+                    newRawResultsMap[data.series_signature].name = series.name;
+                    newRawResultsMap[data.series_signature].platform = series.platform;
+                  });
+                });
+              })).then(function () {
+                $scope.dataLoading = false;
+                displayResults(rawResultsMap, newRawResultsMap);
+              });
+            });
+          });
+        }
+      );
+    }
+
+    //TODO: put this into a generic library
+    function calculateStats(perfData, resultSetID) {
+      var geomeans = [];
+      var total = 0;
+      perfData.forEach(function(pdata) {
```

Comment:
Actually now that I think about it we could express this even more concisely with underscore.js:

_.where(perfData, { result_set_id: resultSetId }).forEach(function(pdata) {
...
}

http://underscorejs.org/#where